### PR TITLE
EOS-18569: BE: Use boto3 package instead of boto package

### PR DESCRIPTION
**Pull request by avrnv**
# Backend

## Problem Statement
Story Ref (if any): [EOS-18569](https://jts.seagate.com/browse/EOS-18569)
## Unit testing on RPM done
Yes
## Problem Description
At the beginning of CORTX CSM team chose boto2 for handling CORTX-specific requests to IAM server (e.g. CreateAccount, CreateAccountLoginProfile, etc.). Per architects' query the dependency on boto2 should be removed.
## Solution
- Implement a method for sending arbitrary AWS-signed requests leveraging boto3's SignV4.
- Implement a new parser for IAM server's responses (the old one leveraged boto.XMLParser).
## Unit Test Cases
- csm/test/plans/s3.pln:test_s3

### CODE REVIEW

**Review by tb-s — APPROVED** [review]

**Comment by avrnv** [comment]
VM Deployment failed in post_install of Motr. Not related to the contents of this PR.

File: csm/plugins/cortx/s3.py
Comment (pawankumarsrivastava) [inline csm/plugins/cortx/s3.py:111]: Instead of `e` please use a relevant variable name

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/plugins/cortx/s3.py
Comment (tb-s) [reply csm/plugins/cortx/s3.py:111]: Are we seriously blocking a PR because a variable name in a 5-line function body is not "relevant"?

See http://catb.org/jargon/html/B/bikeshedding.html .

Comment (pawankumarsrivastava) [reply csm/plugins/cortx/s3.py:111]: @tb-s In the past similar kind of review comments were given by reviewers/architects for the variable names. I personally have corrected proper variable names just because the suggested was more relevant to that function/code.
3-4 days before I encountered a code that says `(a==b)`. If we allow 'e` then we have to allow all such occurrences. 
In one of the functions instead of `e` `element` is used as variable.

Comment (kupranay) [reply csm/plugins/cortx/s3.py:111]: @tb-s  however blocking this PR have multiple reasons like approval of getting new library added, and a thorough testing on a fresh deployed VM is required and tested by QA before we merge it,
 but this variable name not proper itself is more than enough to block the PR.

Comment (avrnv) [reply csm/plugins/cortx/s3.py]: Updated the name.

Comment (dmitrydb) [reply csm/plugins/cortx/s3.py]: @kupranay , Reader can't treat `e` like `event` for one reason: right part of assignment named `Element`. Such code review requests are overhead here.

Comment (avrnv) [reply csm/plugins/cortx/s3.py]: I removed the whole parser since Ujjwal approved xmltodict as a dependency.

**Review by pawankumarsrivastava — APPROVED** [review]
Approving as QA passed

File: csm/plugins/cortx/s3.py
Comment (kupranay) [reply csm/plugins/cortx/s3.py]: @dmitrydb  code readability as important as much as code correctness. I dont think this is overhead. This is good coding practice.

**Review by kupranay — APPROVED** [review]

**Comment by ashish-seagate** [comment]
Tested the changes and observed the Slowness of s3 account and reported it.
Re-tested the build after PR was updated by Dev.
Sanity test suite passed on this build along with other manual test cases for features like Add, edit, delete S3 and CSM users.
Sanity test suite link:
http://eos-jenkins.colo.seagate.com/job/CSM/job/CSM_ROBOT_JOB/129/

**Pull request merged**
